contracts/strategies/BaseStrategy.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import "../core/Allo.sol";
+import "../interfaces/IStrategy.sol";
+
+abstract contract BaseStrategy is IStrategy {
+    /// ======================
+    /// ======= Errors =======
+    /// ======================
+
+    error BaseStrategy_UNAUTHORIZED();
+    error BaseStrategy_STRATEGY_ALREADY_INITIALIZED();
+    error BaseStrategy_INVALID_ADDRESS();
+
+    /// ======================
+    /// ======= Events =======
+    /// ======================
+
+    event Initialized(address allo, bytes32 identityId, uint256 poolId, bytes data);
+
+    /// ==========================
+    /// === Storage Variables ====
+    /// ==========================
+
+    bytes32 internal STRATEGY_IDENTIFIER;
+
+    bytes32 internal identityId;
+    Allo internal allo;
+
+    uint256 internal poolId;
+    bool public initialized;
+
+    /// ====================================
+    /// =========== Modifier ===============
+    /// ====================================
+
+    /// @notice Modifier to check if the caller is the Allo contract
+    modifier onlyAllo() {
+        if (msg.sender != address(allo) && address(allo) != address(0)) {
```

Comment:
I don't think the second half of this check is necessary. If `msg.sender == address(allo)` then it can't be address(0).